beaker-vue/src/util/annotations.ts

```diff
@@ -0,0 +1,364 @@
+import type { Extension } from "@codemirror/state";
+import { EditorView, Decoration, type DecorationSet, hoverTooltip, gutter, GutterMarker } from "@codemirror/view";
+import { linter, lintGutter, type Diagnostic } from "@codemirror/lint";
+import { StateField, StateEffect, RangeSet } from "@codemirror/state";
+
+
+// copied Severity from codemirror-lint, which does not export it
+type LintSeverity = "hint" | "info" | "warning" | "error";
+
+type BeakerLintSeverity = "minor" | "warning" | "info";
+
+function mapBeakerSeverityToLintSeverity(severity: BeakerLintSeverity): LintSeverity {
+    switch (severity) {
+        case 'minor': return 'hint';
```

Comment:
noticed differences between the hidden Severity type on codemirror-lint and our Severity type, so when adding typing had to map to the other when passing our values into the diagnostic severity attribute.